src/yggdrasil/search.go

```diff
@@ -185,13 +228,18 @@ func (s *searches) newIterSearch(dest *crypto.NodeID, mask *crypto.NodeID, callb
 // Otherwise return false.
 func (sinfo *searchInfo) checkDHTRes(res *dhtRes) bool {
 	from := dhtInfo{key: res.Key, coords: res.Coords}
-	if *from.getNodeID() != sinfo.visited && dht_ordered(&sinfo.dest, from.getNodeID(), &sinfo.visited) {
-		// Closer to the destination, so update visited
-		sinfo.searches.router.core.log.Debugln("Updating search:", &sinfo.dest, from.getNodeID(), sinfo.send, sinfo.recv)
-		sinfo.visited = *from.getNodeID()
-		sinfo.time = time.Now()
-	}
 	them := from.getNodeID()
+	sinfo.closest = append(sinfo.closest, them)
+	sort.SliceStable(sinfo.closest, func(i, j int) bool {
```

Comment:
Go’s `container/heap` package may provide some optimisation here rather than repeating sorts.

Reply:
Yes, that could be a good option. I changed the approach I'm using in this PR, so it's not relevant to this spot in the code anymore, but maybe in other places where things are sorted.